var/spack/repos/builtin/packages/podio/package.py

```diff
@@ -131,10 +150,12 @@ class Podio(CMakePackage):
     depends_on("py-tabulate", type=("run", "test"), when="@0.16.6:")
 
     conflicts("+sio", when="@:0.12", msg="sio support requires at least podio@0.13")
+    conflicts("+rntuple", when="@:0.16", msg="rntuple support requires at least podio@0.17")
```

Comment:
Is there a reason to use a conflict instead of variant when? Same for sio, actually.

Reply:
Mainly just out of habit, and my feeling that variants with `when` sometimes seem to confuse the concretizer more than `conflicts`. (But this is really just a feeling, I have literally nothing to back this up and the issues I are more than likely completely unrelated).

Reply:
Ok, sounds good, and I don't see any reason to change it then (at least the conflict leads to a specific error message, not something generic).